internal/provider/data_source_vpc_router_route.go

```diff
@@ -0,0 +1,241 @@
+							"- `vpc`: Name of the VPC " +
+							"- `subnet`: Name of the VPC subnet " +
+							"- `instance`: Name of the instance" +
+							"- `ip`: IP address " +
+							"- `internet_gateway`: name of the internet gateway",
+						Computed: true,
+					},
+				},
+			},
+			"timeouts": timeouts.Attributes(ctx),
+			"time_created": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this VPC router route was created.",
+			},
+			"time_modified": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this VPC router route was last modified.",
+			},
+		},
+	}
+}
+
+func (d *vpcRouterRouteDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state vpcRouterRouteDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readTimeout, diags := state.Timeouts.Read(ctx, defaultTimeout())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, readTimeout)
+	defer cancel()
+
+	params := oxide.VpcRouterRouteViewParams{
+		Route:   oxide.NameOrId(state.Name.ValueString()),
+		Project: oxide.NameOrId(state.ProjectName.ValueString()),
+		Router:  oxide.NameOrId(state.VPCRouterName.ValueString()),
+		Vpc:     oxide.NameOrId(state.VPCName.ValueString()),
+	}
+	route, err := d.client.VpcRouterRouteView(ctx, params)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read VPC router route:",
+			"API error: "+err.Error(),
+		)
+		return
+	}
+	tflog.Trace(ctx, fmt.Sprintf("read VPC router route with ID: %v", route.Id), map[string]any{"success": true})
+
+	// Parse vpcRouterRouteDestinationDataSourceModel into types.Object
+	dm := vpcRouterRouteDestinationDataSourceModel{
+		Type:  types.StringValue(string(route.Destination.Type)),
+		Value: types.StringValue(route.Destination.Value.(string)),
+	}
+	attributeTypes := map[string]attr.Type{
+		"type":  types.StringType,
+		"value": types.StringType,
+	}
+	destination, diags := types.ObjectValueFrom(ctx, attributeTypes, dm)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state.Destination = destination
+
+	// Parse vpcRouterRouteTargetDataSourceModel into types.Object
+	tm := vpcRouterRouteTargetDataSourceModel{
+		Type: types.StringValue(string(route.Target.Type)),
+	}
+
+	// When the target type is set to "drop" the value will be nil
+	if route.Target.Value != nil {
+		tm.Value = types.StringValue(route.Target.Value.(string))
+	}
+
+	targetAttributeTypes := map[string]attr.Type{
+		"type":  types.StringType,
+		"value": types.StringType,
+	}
+	target, diags := types.ObjectValueFrom(ctx, targetAttributeTypes, tm)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state.Target = target
```

Comment:
Same why `types.Object` comment from earlier about the `Target` field.

Reply:
same